conbench/api/runs.py

```diff
@@ -1,26 +1,64 @@
+import datetime
+from typing import Dict, List, Optional, Tuple
+
 import flask as f
 import flask_login
 
 from ..api import rule
 from ..api._docs import spec
 from ..api._endpoint import ApiEndpoint, maybe_login_required
-from ..entities._entity import EntityExists, NotFound
 from ..entities.benchmark_result import BenchmarkResult
 from ..entities.commit import Commit
-from ..entities.run import Run, RunFacadeSchema, RunSerializer
+from ..entities.run import RunFacadeSchema, RunSerializer
+
+
+def get_all_run_info(
+    min_time: datetime.datetime,
+    max_time: datetime.datetime,
+    commit_hashes: Optional[List[str]] = None,
+) -> List[Tuple[BenchmarkResult, int, bool]]:
+    """Get info about each run in a time range.
+
+    Query the database for all benchmark results (optionally filtered to certain commit
+    hashes) in a time range, and return a list of tuples, each corresponding to one
+    run_id. The tuples contain:
+
+    - the earliest BenchmarkResult for that run (by its timestamp)
+    - the number of benchmark results seen for that run
+    - whether any benchmark results are "failed" in that run
+
+    This powers the "list runs" API endpoint and the "recent runs" list on the home page.
+    """
+    run_info: Dict[str, Tuple[BenchmarkResult, int, bool]] = {}  # keyed by run_id
```

Comment:
Hm, I probably should change this into a dataclass. It's pretty similar to the RunForDisplay dataclass in app/index.py. Maybe I can consolidate.